compiler/semexprs.nim

```diff
@@ -1047,7 +1047,7 @@ proc semIndirectOp(c: PContext, n: PNode, flags: TExprFlags; expectedType: PType
       if s != nil:
         setGenericParams(c, n[0])
         return semDirectOp(c, n, flags, expectedType)
-    elif isSymChoice(n[0]):
+    elif isSymChoice(n[0]) and nfDotField notin n.flags:
```

Comment:
I think generic dot field/call will not end up here, which has a special syntax, e.g. `a.len[:int]`. It should be fine.